src/django_peertube_runner_connector/utils/transcoding/hls_playlist.py

```diff
@@ -0,0 +1,138 @@
+""" Functions to update HLS playlists file (.m3u8) of a video. """
+from __future__ import annotations
+
+import logging
+import os
+import re
+
+from django.core.files.uploadedfile import SimpleUploadedFile
+
+import ffmpeg
+
+from django_peertube_runner_connector.models import (
+    Video,
+    VideoFile,
+    VideoStreamingPlaylist,
+)
+from django_peertube_runner_connector.storage import video_storage
+from django_peertube_runner_connector.utils.ffprobe import (
+    get_video_stream_dimensions_info,
+    get_video_stream_duration,
+    get_video_stream_fps,
+)
+from django_peertube_runner_connector.utils.files import (
+    build_file_metadata,
+    get_hls_resolution_playlist_filename,
+    get_video_directory,
+)
+
+from .codecs import get_audio_stream_codec, get_video_stream_codec
+
+
+logger = logging.getLogger(__name__)
+
+UUID_REGEX = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+
+
+def rename_video_file_in_playlist(playlist_path: str, new_video_filename: str) -> None:
+    """Rename the video file in the playlist file content."""
+    with video_storage.open(playlist_path, "r") as playlist_file:
+        content = playlist_file.read()
+
+    new_content = re.sub(
+        f"{UUID_REGEX}-\\d+-fragmented.mp4", new_video_filename, str(content)
+    )
+
+    with video_storage.open(playlist_path, "w") as playlist_file:
+        playlist_file.write(new_content)
+
+
+def on_hls_video_file_transcoding(video: Video, video_file: VideoFile):
+    """
+    Update or create the HLS playlist streaming playlist model for a video.
+    Then it updates / creates its HLS playlist file (.m3u8)
+    """
+    try:
+        playlist = VideoStreamingPlaylist.objects.get(video=video)
+    except VideoStreamingPlaylist.DoesNotExist:
+        playlist = VideoStreamingPlaylist.objects.create(
+            video=video,
+        )
```

Comment:
you can use `get_or_create` ? https://docs.djangoproject.com/en/4.2/ref/models/querysets/#get-or-create